Client/Frontend/Browser/Search/OpenSearch.swift

```diff
@@ -21,6 +21,23 @@ class OpenSearchEngine: NSObject, NSSecureCoding {
     static let defaultSearchClientName = "brave"
     
     let shortName: String
+    
+    // Backwards compatibility workaround, see #3056.
+    // We use `shortName` to store persist what engines are set as default, order etc.
+    // This means there's no easy way to change display text for the search engine without
+    // saved engines breaking.
+    // This updates the engines name in the UI, without changing it at the xml level.
+    // In the future we might refactor it.
+    var displayName: String {
```

Comment:
Do you think it might be better to have an enum for engine types and create a struct for holding the names(shortname/ displayName) and return their rawValue according to Engine type?

But if this is a quick workaround for the problem, I totally get it.

Reply:
Not for now, but I agree, we could have a struct or an enum of search engines we use.
And we should distinguish between display name and search engine identifier, which is not the case now